pkg/diagnose/connectivity/trace_request.go

```diff
@@ -24,10 +24,74 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/config"
 	"github.com/DataDog/datadog-agent/pkg/config/resolver"
 	"github.com/DataDog/datadog-agent/pkg/config/utils"
+	"github.com/DataDog/datadog-agent/pkg/diagnose/diagnosis"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 	"github.com/DataDog/datadog-agent/pkg/util/scrubber"
 )
 
+func init() {
+	diagnosis.Register("connectivity-datadog-core-endpoints", diagnose)
+}
+
+func diagnose(diagCfg diagnosis.DiagnoseConfig) []diagnosis.Diagnosis {
+
+	// Create domain resolvers
+	keysPerDomain, err := utils.GetMultipleEndpoints(config.Datadog)
+	if err != nil {
+		return []diagnosis.Diagnosis{
+			{
+				Result:      diagnosis.DiagnosisSuccess,
+				Name:        "Endpoints configuration",
+				Diagnosis:   "Misconfiguration of agent endpoints",
+				Remediation: "Please validate Agent configuration",
+				RawError:    err,
+			},
+		}
+	}
+
+	diagnoses := make([]diagnosis.Diagnosis, 0)
```

Comment:
```suggestion
	var diagnoses []diagnosis.Diagnosis
```
any reason not to just define this as a var if it has a zero length?

Reply:
Only my ignorance. I am go newbie. I thought that slice need to be explicitly initialized otherwise it would be nil and panic if appended to. Tested your suggestion in Go playground and works well. Thank you.

Reply:
I have pushed the fix BTW